src/url-loader/fs-url-resolver.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as pathlib from 'path';
+import {posix as posix} from 'path';
+import {format as urlLibFormat, Url} from 'url';
+import Uri from 'vscode-uri';
+
+import {parseUrl} from '../core/utils';
+import {FileRelativeUrl, PackageRelativeUrl, ScannedImport} from '../index';
+import {ResolvedUrl} from '../model/url';
+
+import {UrlResolver} from './url-resolver';
+
+const isWindows = process.platform === 'win32';
+
+/**
+ * Resolves package-relative urls to a configured root directory.
```

Comment:
Can you describe this a little more? Does this do any mapping? What's it do with URLs that are already `file:`?

Reply:
clarified